docs/TLS-CIPHERS.md

```diff
@@ -0,0 +1,440 @@
+IANA cipher suite name based on the following schema:
+- prefix is either SSL_ or TLS_ depending on protocol
+- key exchange/agreement algorithm and keys authentication algorithm;
+  client and server exchange public encryption keys (asymmetric encryption)
+  and may authenticate the other side based on certificate. First part of
+  handshake exchanges keys for asymmetric encryption algorithm and then generate
+  key for data encryption by symmetric algorithms (see below). Authentication
+  algorithm verifies that peer public key belongs to the expected other side of
+  communication, i.e. it is not sent by a third party to intercept communication
+  (e.g. man-in-the-middle). Authentication can be static based on certificate and
+  its signature, or dynamic like Kerberos protocol and similar.
+- word _WITH_ separates assymetric key algorithm name from data encryption
+  algorithm (symmentric encryption)
+- message encryption algorithm (symmetric encryption)
+- message hashing algorithm to ensure message integrity, Message Authentication
+  Code (MAC)
+You can read more details about the naming schema here:
+https://docs.microsoft.com/en-us/windows/win32/secauthn/cipher-suites-in-schannel
+Please pay attention, cipher names shown there are specific to Microsoft SChannel
+and do not match exactly to IANA.
+
+column # 1 - 'Number' - IANA Cipher number. 
+column # 2 - 'SSL 3.0 name' - Cipher suite names as they were defined by SSL 3.0
+             RFC document.
+column # 3 - 'TLS IANA name' - cipher suite names as they defined in IANA TLS
+             ciphers registry
+             https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml
+             First TLS ciphers match by name and number to SSL ciphers. They
+             were carried from SSL protocol
+columns ## 4-7 - cipher names implemented by SSL/TLS libraries. Same cipher suite
+             have different name in TLS libraries. Those columns do not specify
+             version of the library which implements cipher, it focus only on
+             names mapping between IANA name and library name.
+column # 8 - 'MacOS SDK' - version of MacOS SDK when cipher suite was first
+             implemented. Ciphers with numbers 0x00 to 0x1D also marked with
+             additional plus and minus signs to show if a special enum value
+             is defined for both SSL and TLS cipher names starting SDK 10.8.
+             More details on this below.
+columns ## 9-12 - protocols introduced by corresponding TLS protocol standard
+             document. More details below.
+column #13 - 'Notes' - some notes about cipher suite
+
+
+| Cipher | SSL 3.0 name                           | TLS IANA name                                 | OpenSSL cipher name           | wolfSSL cipher name            | GSKit cipher name | Network Security                      | MacOS    | TLS 1.0     | TLS 1.1     | TLS 1.2     | TLS 1.3    | Notes |
```

Comment:
Is anyone else able to read this table? My Markdown reader (Typora) can't make any sense of it.

Reply:
Probably table is too wide. I don't know if it can be solved unless broken down to several tables.
I looked at the result only in browser, hope the following link works https://github.com/curl/curl/pull/6464/files?short_path=1c41cba
Horizontal scrolling is very inconvenient. I am open to any suggestion.